routing/routing_helpers.cpp

```diff
@@ -33,6 +33,16 @@ void ReconstructRoute(IDirectionsEngine * engine, IRoadGraph const & graph,
   if (engine)
     engine->Generate(graph, path, times, turnsDir, junctions, trafficSegs, cancellable);
 
+  if (cancellable.IsCancelled())
+    return;
+
+  // In case of any errors in IDirectionsEngine::Generate() |junctions| is empty.
+  if (junctions.empty())
```

Comment:
It does not return anything, isn't it? How you derived that in case of errors in directions engine junctions is empty, if we assume that directions engine works without errors?